cmd/nerdctl/images.go

```diff
@@ -143,19 +146,47 @@ func imagesBashComplete(clicontext *cli.Context) {
 	bashCompleteImageNames(clicontext)
 }
 
-func unpackedImageSize(ctx context.Context, clicontext *cli.Context, client *containerd.Client, i images.Image) (int64, error) {
+type snapshotKey map[string]string
+
+// recursive function to calculate total usage of ChainID's parent
+func (keys snapshotKey) add(ctx context.Context, s snapshots.Snapshotter, usage *snapshots.Usage, key string) error {
+	if key == "" {
+		return nil
+	}
+	u, err := s.Usage(ctx, key)
+	if err != nil {
+		return err
+	}
+	usage.Add(u)
+	return keys.add(ctx, s, usage, keys[key])
+}
+
+func unpackedImageSize(ctx context.Context, clicontext *cli.Context, client *containerd.Client, s snapshots.Snapshotter, i images.Image) (int64, error) {
 	img := containerd.NewImage(client, i)
 
 	diffIDs, err := img.RootFS(ctx)
 	if err != nil {
 		return 0, err
 	}
+
 	chainID := identity.ChainID(diffIDs).String()
-	s := client.SnapshotService(clicontext.String("snapshotter"))
 	usage, err := s.Usage(ctx, chainID)
 	if err != nil {
 		return 0, err
 	}
 
+	keys := make(snapshotKey, 0)
+	walkFn := func(ctx context.Context, info snapshots.Info) error {
+		keys[info.Name] = info.Parent
+		return nil
+	}
+	if err := s.Walk(ctx, walkFn); err != nil {
+		return 0, err
+	}
```

Comment:
Why does it scan all snapshots? Can we walk snapshots only in the chain? I think we can do this by following parents  from the topmost snapshot using `Stat` API.

Reply:
@ktock thank you,  This is what I was looking for ==> `Stat` API . I made modification.